constants.js

```diff
@@ -40,4 +40,21 @@ export const tokensHost = `https://${chainName}all-tokens.webaverse.com`;
 export const landHost = `https://${chainName}sidechain-land.webaverse.com`;
 export const web3MainnetSidechainEndpoint = 'https://mainnetsidechain.exokit.org';
 export const web3TestnetSidechainEndpoint = 'https://testnetsidechain.exokit.org';
-export const homeScnUrl = `https://webaverse.github.io/street/street.scn`;
+export const homeScnUrl = 'https://webaverse.github.io/street/street.scn';
+
+// Setup dialog URL, use hash if on githubpreview and its an okay hash
+const rGithub = /.githubpreview.dev$/;
+const isGH = rGithub.test(window.location.hostname);
+
+const hashHost =
+  /^[a-z][a-z0-9+\-.]*:\/\/([a-z0-9\-._~%!$&'()*+,;=]+@)?([a-z0-9\-._~%]+|\[[a-z0-9\-._~%!$&'()*+,;=:]+\])/.exec(
+    window.location.hash.slice(1),
+  );
+
+const isOKHASH = hashHost && hashHost[2] && rGithub.test(hashHost[2]);
+
+export const dialogUrl = isGH
+  ? isOKHASH
+    ? hashHost[2]
+    : window.location.hostname.replace('3000', '4443')
```

Comment:
This would be better using a regex. We don't want to corrupt domains like `robot3000.club`.

Reply:
it could affect subdomains of the githubpreview.dev only which seems unlikely. We're relying on an undocumented naming convention is the bigger issue imo.